### PR TITLE
Make sure that Keys / Values are memory safe from levelDB

### LINKED ISSUE #108: User can't spend X-Chain funds after ExportUser/ImportUser
**Describe the bug**
If you do `keystore.exportUser` and then `keystore.importUser`, the user doesn't have their X-Chain funds/UTXOs.

**To Reproduce**
* Create a keystore user
* Create an X-chain address
* Send funds to X-Chain address using faucet
* Use `keystore.exportUser` to export the user
* Stop the node. Start again with a different (new) database.
* Use `keystore.importUser` to import the user (output of `keystore.exportUser`)
* Try to send funds from the X-Chain address. Will get "insufficient funds" error. 

**Expected behavior**
A keystore user that is exported then imported has all the same keys, etc. after it is exported/imported as before

**Operating System**
Ubuntu 18.04

**Additional context**
Thanks to @michaelbnewman for finding this bug 

By submitting this issue I agree to the [Terms and Conditions](https://files.avalabs.org/legal/bug_hunt_q2_2020_terms.pdf) of the [Developer Accelerator Program](https://avalabs.org/dap).

### PR DESCRIPTION
**Pull request by StephenButtolph**
This adds a DB test that ensures that iterators don't overwrite key / value byte arrays returned from the iterator. This resolves #108

### CODE REVIEW

**Comment by danlaine** [comment]
@StephenButtolph  Does this also resolve #123 ?

File: database/leveldb/db.go [hunk 1]
```diff
@@ -220,3 +227,13 @@ func updateError(err error) error {
 		return err
 	}
 }
+
+func copyBytes(bytes []byte) []byte {
```
Comment (danlaine) [inline database/leveldb/db.go:231]: We already have a `copyBytes` function in `database/encdb/db.go`
Let's move this to `utils`

**Comment by StephenButtolph** [comment]
> @StephenButtolph Does this also resolve #123 ?

I'm not really sure, as that issue hasn't been repeatable for me. However, if the root cause is the same as #108, which stems from `exportUser`, then this could resolve that.

File: database/test_database.go [hunk 2]
```diff
@@ -622,6 +623,63 @@ func TestIteratorStartPrefix(t *testing.T, db Database) {
...
+		t.Fatalf("Unexpected error on batch.Put: %s", err)
+	} else if err := db.Put(key2, value2); err != nil {
+		t.Fatalf("Unexpected error on batch.Put: %s", err)
+	} else if err := db.Put(key3, value3); err != nil {
+		t.Fatalf("Unexpected error on batch.Put: %s", err)
+	}
+
+	iterator := db.NewIterator()
+	if iterator == nil {
+		t.Fatalf("db.NewIteratorWithStartAndPrefix returned nil")
```
Comment (danlaine) [inline database/test_database.go:647]: Should say "db.NewIterator" is nil

File: database/leveldb/db.go (hunk 1 above)
Comment (StephenButtolph) [reply database/leveldb/db.go]: done

File: database/test_database.go (hunk 2 above)
Comment (StephenButtolph) [reply database/test_database.go]: updated

**Pull request merged**